src/main/java/com/curity/mongodb/datasource/MongoUserAccountDataAccessProvider.java

```diff
@@ -0,0 +1,141 @@
+    }
+
+    @Override
+    public ResourceAttributes<?> getByPhone(String s, ResourceQuery.AttributesEnumeration attributesEnumeration)
+    {
+        return _mongoUtils.getAccountAttributes("phoneNumbers", s, true);
+    }
+
+    @Override
+    public AccountAttributes create(AccountAttributes accountAttributes)
+    {
+        Document document = new Document(accountAttributes.toMap());
+        _database.getCollection(USERS_COLLECTION).insertOne(document);
+
+        AccountAttributes newAccountAttributes = _mongoUtils.getAccountAttributes("userName",
+                accountAttributes.getUserName(), false);
+        newAccountAttributes = newAccountAttributes.removeAttribute("password");
+        return newAccountAttributes;
+    }
+
+    @Override
+    public ResourceAttributes<?> update(AccountAttributes accountAttributes,
+                                        ResourceQuery.AttributesEnumeration attributesEnumeration)
+    {
+        _database.getCollection(USERS_COLLECTION).updateOne(eq("userName", accountAttributes.getUserName()),
+                new Document("$set", new Document(accountAttributes.toMap())));
+        AccountAttributes newAccountAttributes = _mongoUtils.getAccountAttributes("userName",
+                accountAttributes.getUserName(), false);
+
+        return filterAttributes(newAccountAttributes.toMap(), attributesEnumeration);
+    }
+
+    @Override
+    public ResourceAttributes<?> update(String s, Map<String, Object> map,
+                                        ResourceQuery.AttributesEnumeration attributesEnumeration)
+    {
+        _database.getCollection(USERS_COLLECTION).updateOne(eq("userName", s),
+                new Document("$set", new Document(map)));
+        AccountAttributes newAccountAttributes = _mongoUtils.getAccountAttributes("userName", s, false);
+
+        return filterAttributes(newAccountAttributes.toMap(), attributesEnumeration);
+    }
+
+    @Override
+    public ResourceAttributes<?> patch(String s, AttributeUpdate attributeUpdate,
+                                       ResourceQuery.AttributesEnumeration attributesEnumeration)
+    {
+        Map<String, Object> dataMap = attributeUpdate.getAttributeReplacements().toMap();
+        dataMap.putAll(attributeUpdate.getAttributeAdditions().toMap());
+        _database.getCollection(USERS_COLLECTION).updateOne(eq("userName", s),
+                new Document("$set", new Document(dataMap)));
+        AccountAttributes newAccountAttributes = _mongoUtils.getAccountAttributes("userName", s, false);
+
+        return filterAttributes(newAccountAttributes.toMap(), attributesEnumeration);
+    }
+
+    @Override
+    public void link(String s, String s1, String s2)
+    {
+
+    }
+
+    @Override
+    public void delete(String s)
+    {
+        _database.getCollection(USERS_COLLECTION).deleteOne(eq("userName", s));
+    }
+
+    @Override
+    public ResourceQueryResult getAll(long l, long l1)
+    {
+        return null;
```

Comment:
This is a TODO, right?

Reply:
yes, but I am not sure what to do in this function.